cmake/modules/AddCephTest.cmake

```diff
@@ -1,8 +1,30 @@
 #AddCephTest is a module for adding tests to the "make check" target which runs CTest
 
+macro(set_test_props labels cost cost_default timeout)
+  if("${timeout}" GREATER 0)
+    set_property(TEST ${test_name} PROPERTY TIMEOUT ${timeout})
+  else()
+    set_property(TEST ${test_name} PROPERTY TIMEOUT ${CEPH_TEST_TIMEOUT})
+  endif()
+  if("${cost}" GREATER 0)
+    set_property(TEST ${test_name} PROPERTY COST ${cost})
+  else()
+    set_property(TEST ${test_name} PROPERTY COST ${cost_default})
+  endif()
+  if(NOT "${labels}" EQUAL "")
+    set_property(TEST ${test_name} APPEND PROPERTY LABELS "${labels}")
+  endif()
+endmacro()
+
 #adds makes target/script into a test, test to check target, sets necessary environment variables
 function(add_ceph_test test_name test_path)
-  add_test(NAME ${test_name} COMMAND ${test_path} ${ARGN})
+  set(options "")
+  set(oneValueArgs COST TIMEOUT)
+  set(multiValueArgs LABELS)
```

Comment:
no need to create these variables. and we don't use camelCase in the cmake files.